.github/workflows/smithy-dafny-conversion.yml

```diff
@@ -11,10 +11,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
       - uses: actions/setup-java@v3
         with:
           distribution: 'corretto'
-          java-version: '8'
+          java-version: '17'
```

Comment:
Issue/Blocking: Revert this.

The Conversion Library is a runtime dependency of products that MUST support Java 8.
As such, it is built and tested with Java 8.

The code generators are not runtime dependencies,
and use the latest and greatest Java features.